cmake/zig-build.cmake

```diff
@@ -103,6 +273,36 @@ add_custom_target(zig_build
     BYPRODUCTS ${CMAKE_BINARY_DIR}/lib/libapp_zig.a
     VERBATIM)
 
+add_dependencies(zig_build translate_c)
+add_library(app_zig STATIC IMPORTED)
+set_target_properties(app_zig PROPERTIES IMPORTED_LOCATION ${CMAKE_BINARY_DIR}/lib/libapp_zig.a)
+add_dependencies(app_zig zig_build)
+
 add_prebuilt_library(zig ${CMAKE_BINARY_DIR}/lib/libapp_zig.a)
-add_dependencies(zig zig_build)
+add_dependencies(${COMPONENT_LIB} zig_build)
+
+
 target_link_libraries(${COMPONENT_LIB} PRIVATE ${CMAKE_BINARY_DIR}/lib/libapp_zig.a)
+
+target_link_libraries(${COMPONENT_LIB} PRIVATE
+    "-Wl,--start-group"
+    ${IDF_PATH}/components/esp_phy/lib/${TARGET_IDF_MODEL}/libphy.a
```

Comment:
h2/p4: `ninja: error: '/opt/esp/idf/components/esp_phy/lib/esp32p4/libphy.a', needed by 'zig-sample-idf.elf', missing and no known rule to make it`